tests/test_assembly.py

```diff
@@ -0,0 +1,66 @@
+import json
+import subprocess
+from pathlib import Path
+from typing import TypedDict
+
+import pytest
+import yaml
+from pydantic import RootModel
+
+from unicon_backend.evaluator.problem import Problem
+from unicon_backend.evaluator.tasks.base import TaskType
+from unicon_backend.evaluator.tasks.programming.base import RequiredInput
+from unicon_backend.runner import RunnerProgram
+
+
+def execute_program(program: RunnerProgram, temp_dir: Path) -> tuple[int, str, str]:
```

Comment:
I am of the opinion that it's probably better to do snapshot testing on the generated CST instead of comparing execution results. With snapshot testing, we don't need an extra harness for code execution and we can ensure **assembly** is structurally equivalent during regression testing. Additionally, there could be cases where 2 different CSTs produces the same execution results which may not ideal when we want to catch regressions to our transform passes e.g. import hoisting

Reply:
makes sense! by snapshot testing, do you mean comparing that the code itself (not the output) is an exact match instead?

Reply:
We can reuse the procedure in `unicon_backend.cli` to get the resulting CST for each testcase

Reply:
> makes sense! by snapshot testing, do you mean comparing that the code itself (not the output) is an exact match instead?

yeap, we can probably either compare the output CST or the assembled program

Reply:
refactored to compare assembled program in https://github.com/uniconhq/unicon-backend/pull/55/commits/17dbade41cf0cbdc274a35a4f76218ef3a717f9a